dwl.c

```diff
@@ -2129,18 +2140,15 @@ setup(void)
 {
 	int layer;
 
-	/* Set up signal handlers */
-	struct sigaction sa = {.sa_flags = SA_RESTART, .sa_handler = sigchld};
-	sigemptyset(&sa.sa_mask);
-	sigaction(SIGCHLD, &sa, NULL);
-
-	sa.sa_handler = quitsignal;
-	sigaction(SIGINT, &sa, NULL);
-	sigaction(SIGTERM, &sa, NULL);
-
 	/* The Wayland display is managed by libwayland. It handles accepting
 	 * clients from the Unix socket, manging Wayland globals, and so on. */
 	dpy = wl_display_create();
+	eventloop = wl_display_get_event_loop(dpy);
+	wl_event_loop_add_signal(eventloop, SIGCHLD, handlesig, NULL);
+	wl_event_loop_add_signal(eventloop, SIGINT, handlesig, NULL);
+	wl_event_loop_add_signal(eventloop, SIGTERM, handlesig, NULL);
+	/* Avoid terminating if nobody is reading the status output */
+	wl_event_loop_add_signal(eventloop, SIGPIPE, handlesig, NULL);
```

Comment:
I am not sure if we need to call `wl_event_source_remove`

Reply:
Ah, I think you're right.  It doesn't look like anything in the Wayland code cleans this up.  I'll try to figure out if there's a way to do this "correctly" while keeping delta <= 0.